napari/layers/image/image.py

```diff
@@ -742,7 +753,7 @@ def _set_view_slice(self) -> None:
         # This can happen when slicing layers with different extents.
         indices = self._slice_indices
         for d in self._slice_input.not_displayed:
-            if (indices[d] < 0) or (indices[d] >= self._extent_data[1][d]):
+            if (indices[d] < 0) or (indices[d] >= self._extent_data[1][d] + 1):
```

Comment:
These are both integers right?

```suggestion
            if (indices[d] < 0) or (indices[d] > self._extent_data[1][d]):
```

Reply:
No, extent data is floats (though indeed the image ones are in integer steps). You think it's more readable with `>`?

Reply:
I'd say slightly more readable because I have to do one less mathematical operation. Don't feel strongly though, so I'll leave it up to you. This is effectively an out-of-bounds check, so you could also do `not (0 < indices[d] < self._extent_data[1][d])` if you're feeling fancy.